tinygrad/codegen/linearizer.py

```diff
@@ -391,7 +391,7 @@ def upcast_strides(buf:int):
     self.global_store(0, global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs, val)
 
     # optimize the uops
-    self.uops.uoptimize()
+    self.uops.uoptimize({op: name for name, op in self.loop_uops.items()}, self.render_ops, self)
```

Comment:
This state. It's a lot harder to reason about a program when state like this is passed